pkg/controller/gkenetworkparamset/gkenetworkparamset_controller.go

```diff
@@ -335,6 +335,13 @@ func (c *Controller) syncNetworkWithGNP(ctx context.Context, network *networkv1.
 
 // handleGNPDelete checks to see if its safe to delete the GNP resource before calling executeGNPDelete on it
 func (c *Controller) handleGNPDelete(ctx context.Context, params *networkv1.GKENetworkParamSet) error {
+
+	// GNP controller will wait for it to be the last finalizer to handle deletion
+	isOnlyGNPFinalizer := len(params.Finalizers) == 1 && params.Finalizers[0] == GNPFinalizer
```

Comment:
what if the [0] is not GNPFinalizer, we will never process beyond this and get stuck? we should handle such situation, should we maybe re-add GNP finalizer, WDYT?

Reply:
So this situation could occur if the the network update fails in executeGNPDelete because GNP would've gotten updated with the removed finalizer.

I've updated the reconcile so that we don't update GNP if we run into errors which will prevent this from being a possibility.

Reply:
I am more concern of situation where someone manually removed your finalizer. this will become a deadlock, maybe we should just not check the type, just the count?

Reply:
I see what you mean. Would that not just put us back to the issue we're currently having where ParamsReady is set to False to early?

Reply:
It might, but in 90% cases only controllers will handle those, so the order will be correct and GNP finalizer will be the last one in. the other cases we might get into what you say, but we give Cx ability to manually clean the finalizers as well.